src/app/Fake.DotNet.Testing.VSTest/VSTest.fs

```diff
@@ -0,0 +1,150 @@
+      TestAdapterPath: string}
+
+/// VSTest default parameters.
+let VSTestDefaults = 
+    { SettingsPath = null
+      Tests = []
+      EnableCodeCoverage = false
+      InIsolation = true
+      UseVsixExtensions = false
+      Platform = null
+      Framework = null
+      TestCaseFilter = null
+      Logger = null
+      ListTestsPath = null
+      ListDiscoverers = false
+      ListExecutors = false
+      ListLoggers = false
+      ListSettingsProviders = false
+      ToolPath = 
+          match Process.tryFindFile vsTestPaths vsTestExe with
+          | Some path -> path
+          | None -> ""
+      WorkingDir = null
+      TimeOut = TimeSpan.MaxValue
+      ErrorLevel = ErrorLevel.Error
+      TestAdapterPath = null }
+
+/// Builds the command line arguments from the given parameter record and the given assemblies.
+/// [omit]
+let buildVSTestArgs (parameters : VSTestParams) assembly = 
+    let testsToRun = 
+        if not (Seq.isEmpty parameters.Tests) then 
+            sprintf @"/Tests:%s" (parameters.Tests |> String.separated ",")
+        else null
+    new StringBuilder()
+    |> StringBuilder.appendIfTrue (assembly <> null) assembly
+    |> StringBuilder.appendIfNotNull parameters.SettingsPath "/Settings:"
+    |> StringBuilder.appendIfTrue (testsToRun <> null) testsToRun
+    |> StringBuilder.appendIfTrue parameters.EnableCodeCoverage "/EnableCodeCoverage"
+    |> StringBuilder.appendIfTrue parameters.InIsolation "/InIsolation"
+    |> StringBuilder.appendIfTrue parameters.UseVsixExtensions "/UseVsixExtensions:true"
+    |> StringBuilder.appendIfNotNull parameters.Platform "/Platform:"
+    |> StringBuilder.appendIfNotNull parameters.Framework "/Framework:"
+    |> StringBuilder.appendIfNotNull parameters.TestCaseFilter "/TestCaseFilter:"
+    |> StringBuilder.appendIfNotNull parameters.Logger "/Logger:"
+    |> StringBuilder.appendIfNotNull parameters.ListTestsPath "/ListTests:"
+    |> StringBuilder.appendIfTrue parameters.ListDiscoverers "/ListDiscoverers"
+    |> StringBuilder.appendIfTrue parameters.ListExecutors "/ListExecutors"
+    |> StringBuilder.appendIfTrue parameters.ListLoggers "/ListLoggers"
+    |> StringBuilder.appendIfTrue parameters.ListSettingsProviders "/ListSettingsProviders"
+    |> StringBuilder.appendIfNotNull parameters.TestAdapterPath "/TestAdapterPath:"
+    |> StringBuilder.toText
+
+/// Runs VSTest command line tool (VSTest.Console.exe) on a group of assemblies.
+/// ## Parameters
+/// 
+///  - `setParams` - Function used to manipulate the default VSTestParams values.
+///  - `assemblies` - Sequence of one or more assemblies containing Microsoft Visual Studio Unit Test Framework unit tests.
+/// 
+/// ## Sample usage
+///
+///     Target.create "Test" (fun _ ->
+///         !! (testDir + @"\*.Tests.dll") 
+///           |> VSTest.VSTest (fun p -> { p with SettingsPath = "Local.RunSettings" })
+///     )
+let VSTest (setParams : VSTestParams -> VSTestParams) (assemblies : string seq) = 
```

Comment:
Maybe rename that such that it looks like `VSTest.run` for example (look how other runners call it).